### ISSUE TITLE
Data for Mongolia

### ISSUE THREAD

**Issue by q--** [electricitymaps/electricitymaps-contrib#3911]
TL;DR parser-buildable [generation data for Mongolia](https://ndc.energy.mn/?page_id=420); only (the most important) thermal power plants are broken down + some unknown generators, unsure if other generation is included in the remainder or simply missing.

----

I went looking if I could find any data on Mongolia.

The good news: I found some real-time generation data. The bad news: it appears to be rather incomplete, though I might be overlooking things because I don't speak Mongolian.

The best information I could find was https://ndc.energy.mn/?page_id=420

This page has real-time generation for the current? hour of today, and for the same time last year, in [this bar chart](https://peak.energy.mn/bar.php)

Translation of the names according to Google:

- TPP-4
- TPP-3
- TPP-2
- DDSS
- EDCS
 
The top ones are the power plants with the highest installed capacity in the country according to [a report by the Nautilus Institute](https://nautilus.org/napsnet/napsnet-special-reports/mongolian-grid-data/) ([PDF version](https://nautilus.org/wp-content/uploads/2020/09/Tovuudorj-Purevjav-Grid-Data-SR-PDF-Sep20-2020.pdf)). Mongolia doesn't have that many power plants, so this covers most generation. Tip, you can find the power plant by Ctrl+F searching for the Mongolian name on https://www.energy.mn/. DDSS = [Darkhan Thermal Power Plant](http://dpp.energy.mn/)  and EDCS [Erdenet Thermal Power Plant](http://erdpp.energy.mn/).  
As an aside, the Mongolian grid [appears](https://nautilus.org/wp-content/uploads/2020/09/Fig1-2.jpg) to consist of smaller grids that have limited (if any?) interconnectivity. Since the data I found mentions the capital, this data at least encompasses the Central grid.

Then there is [what appears to be a load curve graph](https://peak.energy.mn/chart.php), but the blue-green "Mode planning" looks rather weird. The values on the red line seem to add up to a approximately the generation from the bar chart mentioned above, so that looks realistic. However, the title translates to "Daily load information", so it might actually be consumption

Labels:

- Горим төлөвлөлт = Mode planning - possibly the generation (or demand) forecast
- Гүйцэтэл = Done (or 'That's it') - possibly the actual generation (or demand)

Том хот суурингийн ачаалал

There's also the [Том хот суурингийн ачаалал "The load of a large city (or: 'large urban congestion')"](https://peak.energy.mn/jspie.php?reload=1) chart. Right now it only has 'Улаанбаатар', which is Ulaanbaatar, the capital. The number given appears to be lower than that of the bars in the bar chart or the current hour in the graph.

Lastly, I found [this chart](https://peak.energy.mn/pie.php) via Google on the same subdomain. Their names correspond to those of Mongolia's three wind power plants [in this table from the Nautilus report](https://nautilus.org/wp-content/uploads/2020/09/Table1-2.jpg); Google translation of the names on the web page:

- Salkhit (SCS or STS or SSC)
- Tsetsii (SCS or STS or SSC)
- Shand (SCS or STS or SSC)

Some of the power plants have their own web page with more information, see the last few links in the footer. See also the nautilus document I linked earlier.

Side note, I suspect the login-protected sites https://dsums.transco.mn/ and/or http://data.transco.mn/  might have more detailed information. But perhaps not, the description on their login page (e.g. "Integrated technical database") could very well mean they're more useful for e.g. maintenance technicians than people looking for statistics.

**Comment by alixunderplatz** [comment]
I had a look at the Mongolian 🇲🇳🇲🇳🇲🇳 data, let's make a parser by using the most recent hourly data from the website @q--  mentioned above.

https://ndc.energy.mn

Here's the data of the frame with the desired datapoints in the HTML:
https://disnews.energy.mn/test/convert.php

<img width="814" alt="image" src="https://user-images.githubusercontent.com/25743609/215353753-40e4d503-77b5-47d9-9860-55e64fdc78ef.png">

total load | solar energy | wind energy | import | air temperature

So we can calculate the remaining production as follows:
coal = total load - solar - wind - import
(oil and hydro make up less than 1% of total yearly production, so it is >99% coal)

Timestamp and date are available as well.

Let's make it happen and get another big zone on the map! 🇲🇳

**Comment by alixunderplatz** [comment]
@VIKTORVAV99 working on this parser right now :)

5:00 local time: 
`
{'coal': 901.8699999999999, 'solar': 46.82, 'wind': 67.69}
`

6:00 local time: 
`
fetch_production() ->
{'zoneKey': 'MN', 'datetime': datetime.datetime(2023, 1, 31, 6, 0, tzinfo=tzfile('/usr/share/zoneinfo/Asia/Ulaanbaatar')), 'production': {'coal': 913.82, 'solar': 45.4, 'wind': 66.27}, 'storage': {}, 'source': 'https://ndc.energy.mn/'}
`

The data for the hour 06:00 was available right at 07:00 local time, so it is the most recent production of the previous hour. It is indeed hourly data. At nighttime after 01:00 there might be a longer lag for the next data point, but we will see once the parser is up and running, I guess :-)

I'll likely finish this one by tomorrow, will keep you updated 😺

**Comment by VIKTORVAV99** [comment]
Are you using the CSV downloads? I found those to be the most reliable when playing around. The source also has historical data but I haven't found the exact endpoint/params for it yet.

**Comment by Nil-Cipher** [comment]
Is this issue still open? I can take a look at it.

**Comment by VIKTORVAV99** [comment]
I actually think this should be closed as we now have a parser for Mongolia.
